client/my-sites/customer-home/cards/features/stats-v2/utils.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Internal dependencies
+ */
+import { getLoadingTabs } from 'state/stats/chart-tabs/selectors';
+import {
+	isRequestingSiteStatsForQuery,
+	getSiteStatsForQuery,
+	getSiteStatsNormalizedData,
+} from 'state/stats/lists/selectors';
+
+export const isLoadingStats = (
+	state,
+	siteId,
+	chartTab,
+	chartPeriod,
+	insightsQuery,
+	topPostsQuery,
+	visitsQuery
+) =>
+	getLoadingTabs( state, siteId, chartPeriod ).includes( chartTab ) ||
+	isRequestingSiteStatsForQuery( state, siteId, 'statsInsights', insightsQuery ) ||
+	isRequestingSiteStatsForQuery( state, siteId, 'statsTopPosts', topPostsQuery ) ||
+	isRequestingSiteStatsForQuery( state, siteId, 'statsVisits', visitsQuery );
+
+export const getMostPopularDatetime = ( state, siteId, query ) => {
```

Comment:
Nonblocking, but I think we can make each of these into global selectors and drop the util file. If some of these are doing intensive work it's also possible to memo values using `lib/create-selector` though no need for the extra complexity unless we're seeing perf issues.

Reply:
https://github.com/Automattic/wp-calypso/pull/41441/commits/7859e65b848da308c594825564978fdb653cb3c8